teamcity: failed tests on master: acceptance/TestDockerMigration_7429

The following tests appear to have failed:

[#228372](https://teamcity.cockroachdb.com/viewLog.html?buildId=228372):

```
--- FAIL: acceptance/TestDockerMigration_7429 (6.420s)
test_log_scope.go:80: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestDockerMigration_7429058687400
test_log_scope.go:61: use -show-logs to present logs inline
reference_test.go:41: non-zero exit code: 1
------- Stdout: -------
test logs left over in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestDockerMigration_7429058687400

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/rpc/context_test.go
+++ b/pkg/rpc/context_test.go
@@ -34,6 +34,7 @@ import (
 
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/util"
+	"github.com/cockroachdb/cockroach/pkg/util/grpcutil"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -627,23 +628,14 @@ func TestGRPCKeepaliveFailureFailsInflightRPCs(t *testing.T) {
 	}
 	defer func() { _ = conn.Close() }()
 
-	// We'll expect any of the errors which tests revealed that the RPC call might
-	// return when an RPC's transport connection is closed because of the
-	// heartbeats timing out.
-	gRPCErrorsRegex := "transport is closing|" +
-		"rpc error: code = Unavailable desc = grpc: the connection is unavailable|" +
-		"rpc error: code = Internal desc = transport: io: read/write on closed pipe|" +
-		"rpc error: code = Internal desc = transport: tls: use of closed connection|" +
-		"rpc error: code = Internal desc = transport: EOF|" +
-		"use of closed network connection"
-
 	// Perform an RPC so that a connection gets opened. In theory this RPC should
 	// succeed (and it does when running without too much stress), but we can't
-	// rely on that - see comment on the timeout above.
+	// rely on that - it's possible that the RPC call could return earlier due to
+	// its transport connection being closed because of heartbeats timing out.
 	heartbeatClient := NewHeartbeatClient(conn)
 	request := PingRequest{}
 	if _, err := heartbeatClient.Ping(context.TODO(), &request); err != nil {
-		if !testutils.IsError(err, gRPCErrorsRegex) {
+		if !grpcutil.IsClosedConnection(err) {
 			t.Fatal(err)
 		}
 		// In the rare eventuality that we got the expected error, this test
@@ -665,8 +657,7 @@ func TestGRPCKeepaliveFailureFailsInflightRPCs(t *testing.T) {
 
 	transportConn.PartitionC2S()
 
-	if _, err := heartbeatClient.Ping(context.TODO(), &request); !testutils.IsError(
-		err, gRPCErrorsRegex) {
+	if _, err := heartbeatClient.Ping(context.TODO(), &request); !grpcutil.IsClosedConnection(err) {
 		t.Fatal(err)
 	}
 

--- a/pkg/util/grpcutil/grpc_util.go
+++ b/pkg/util/grpcutil/grpc_util.go
@@ -17,6 +17,7 @@
 package grpcutil
 
 import (
+	"io"
 	"strings"
 
 	"github.com/cockroachdb/cockroach/pkg/util/netutil"
@@ -50,7 +51,10 @@ func IsClosedConnection(err error) bool {
 		grpc.Code(err) == codes.Unavailable ||
 		grpc.ErrorDesc(err) == grpc.ErrClientConnClosing.Error() ||
 		strings.Contains(err.Error(), "is closing") ||
+		strings.Contains(err.Error(), "tls: use of closed connection") ||
 		strings.Contains(err.Error(), "use of closed network connection") ||
+		strings.Contains(err.Error(), io.ErrClosedPipe.Error()) ||
+		strings.Contains(err.Error(), io.EOF.Error()) ||
 		strings.Contains(err.Error(), "node unavailable") {
 		return true
 	}

